### DISCUSSION TITLE
Blurry SVG path on Firefox when zooming

### DISCUSSION THREAD

**Discussion by lebesnec** [d3/d3#3616] [Q&A] [answered]
I don't know if this is a Firefox bug or a d3 bug or if I'm doing something wrong but I have a very blurry rendering for SVG path when I zoom, in Firefox (Chrome et Edge are working) :

Firefox :
![image](https://user-images.githubusercontent.com/49292/208964219-37922da3-82d1-4d00-88d6-e69e193557c4.png)

Chrome
![image](https://user-images.githubusercontent.com/49292/208964269-5851379b-1e53-478b-9929-e49d2500b698.png)

Demo (try to zoom to the max level on the red circle) :
https://plnkr.co/edit/oBMsUrvDppsPHlhN

**Comment by Fil** [comment, marked as answer]
Can you upgrade the test to a recent version of d3? Current version is 7.8.

In script.js you'll need to change the code to:

~~~js
function zoomed({transform}) {
  layer.attr("transform", transform);
}

var svg = d3.select("#app")
        .append("svg")
        .attr("width", 500)
        .attr("height", 500)
        .call(zoom)

var layer = svg.append("g").attr("transform", transform);

svg.call(zoom.transform, transform);

~~~

**Reply by Fil** [reply]
Note: I see the same thing as you on Firefox. Note sure what to do about it.

**Reply by lebesnec** [reply]
oh yes I forgot to change the version when I forked that example.
Here is an updated version with your recommended changes: https://plnkr.co/edit/xDadopItbHBMzD9w

My guess is that something is wrong with Firefox anti-aliasing, and it's not a problem with d3.js.
